openstack/standalone-cinder/pkg/volumeservice/actions.go

```diff
@@ -108,12 +111,10 @@ func CreateCinderVolume(vs *gophercloud.ServiceClient, options controller.Volume
 // become available.  The connection information cannot be retrieved from cinder
 // until the volume is available.
 func WaitForAvailableCinderVolume(vs *gophercloud.ServiceClient, volumeID string) error {
-	// TODO: Implement proper polling instead of brain-dead timers
-	c := make(chan error)
-	go time.AfterFunc(5*time.Second, func() {
-		c <- nil
+	return wait.Poll(interval, timeout, func() (bool, error) {
+		status, err := GetCinderVolumeStatus(vs, volumeID)
+		return status == "available", err
```

Comment:
i am paranoid but making strings.ToLower() to these strings is more robust

Reply:
that make sense. Fix it in next change